protos/peers.proto

```diff
@@ -70,6 +70,36 @@ message MongoConfig {
   string database = 5;
 }
 
+message AwsAuthStaticCredentialsConfig {
+  string access_key_id = 1 [(peerdb_redacted) = true];
+  string secret_access_key = 2 [(peerdb_redacted) = true];
+}
+
+message AWSAuthAssumeRoleConfig {
+  string assume_role_arn = 1;
+  optional string chained_role_arn = 2;
+}
+
+enum AwsIAMAuthConfigType {
+  IAM_AUTH_AUTOMATIC = 0;
+  IAM_AUTH_STATIC_CREDENTIALS = 1;
+  IAM_AUTH_ASSUME_ROLE = 2;
+}
+
+message AwsAuthenticationConfig {
+  string region = 1;
+  AwsIAMAuthConfigType auth_type = 2;
+  oneof auth_config {
+    AwsAuthStaticCredentialsConfig static_credentials = 3;
+    AWSAuthAssumeRoleConfig role = 4;
+  }
+}
+
+enum PostgresAuthType {
+  POSTGRES_PASSWORD = 0;
```

Comment:
nit: 0 should be reserved for unknown in these enums, wherever applicable

Reply:
I'm still thinking about it, in our case default is always password

Reply:
@serprex what do you think? old configs which do not have this set should automatically start using password

Reply:
0 for password is fine